pkg/vault/test/foundry/mutation/router/Router.t.sol

```diff
@@ -212,4 +215,169 @@ contract RouterMutationTest is BaseVaultTest {
         vm.expectRevert(abi.encodeWithSelector(IVaultErrors.SenderIsNotVault.selector, address(this)));
         router.queryRemoveLiquidityRecoveryHook(pool, msg.sender, 10);
     }
+
+    function testQuerySwapSingleTokenExactInSaveSender() public {
+        HooksConfig memory hooksConfig = vault.getHooksConfig(pool);
+        hooksConfig.shouldCallBeforeSwap = true;
+        vault.manualSetHooksConfig(pool, hooksConfig);
+
+        PoolHooksMock(poolHooksContract).setShouldStoreSavedSender(true);
```

Comment:
Could avoid this line (and the others below), by reversing the sense of the flag (unless you have to set it false in even more places, of course).